funflow/src/Control/Funflow/External.hs

```diff
@@ -179,13 +179,25 @@ instance FromJSON OutputCapture
 instance ToJSON OutputCapture
 instance Store OutputCapture
 
+-- | Control the environment set for the external process. This can either
+--   inherit from the surrounding environment, or explicitly set things.
+data Env
+  = EnvInherit
+  | EnvExplicit [(T.Text, Param)]
+  deriving (Generic, Show)
+
+instance ContentHashable IO Env
```

Comment:
We may want to point out explicitly that `EnvInherit`'s hash is independent of the actual environment, and that a change in environment will not cause a rerun of the task. Can point to the haddocks of `Param` above for that.

Reply:
It would likely be better here to actually hash the environment, I guess...

Reply:
Without some additional black-listing that's likely to cause a fair amount of duplicated work. For example on a cluster different values in `$HOSTNAME` would produce different hashes, and so cached results would never be re-used between different nodes.

For occurrences of `ParamEnv` in `EnvExplicit` (and I guess also command line args) we may want to incorporate the value of the env var into that hash. But, I think it's good to have `EnvInherit` to allow the user to explicitly ignore the environment in the hash.